packages/services/src/kernel/comm.ts

```diff
@@ -199,11 +199,10 @@ export class CommHandler extends DisposableDelegate implements Kernel.IComm {
       buffers
     );
     let future = this._kernel.sendShellMessage(msg, false, true);
-    options.channel = 'iopub';
     let onClose = this._onClose;
     if (onClose) {
       let ioMsg = KernelMessage.createMessage(
-        options,
+        { ...options, msgType: 'comm_close', channel: 'iopub' },
```

Comment:
I suppose, but I erred on the side of consistency here, so that the code would be easier to follow, i.e., I'm overriding msgType and channel everywhere.